src/apps/api/tests/test_users.py

```diff
@@ -0,0 +1,110 @@
+from django.urls import reverse
+from rest_framework.test import APITestCase
+
+from factories import UserFactory, ProducerFactory, EmailAddressFactory, ProfileFactory
+from profiles.models import EmailAddress
+
+
+class ChahubAPITestCase(APITestCase):
+    def login(self, username, password='test'):
+        self.client.logout()
+        self.client.login(username=username, password=password)
+
+
+class TestEmailPermissions(ChahubAPITestCase):
+    def setUp(self):
+        self.admin = UserFactory(username='admin', is_superuser=True)
+        self.user = UserFactory(username='test')
+        self.norm = UserFactory(username='norm')
+
+    def add_email(self, email_address, user=None):
+        user = user or self.user
+        return self.client.post(
+            reverse('user-add_email_address', kwargs={'pk': user.id, 'version': 'v1'}),
+            data={'email_address': email_address}
+        )
+
+    def remove_email(self, email_address, user=None):
+        user = user or self.user
+        email_pk = EmailAddress.objects.get(email=email_address).id
+        return self.client.delete(
+            reverse('user-remove_email_address', kwargs={'pk': user.id, 'version': 'v1'}),
+            data={'email_pk': email_pk}
+        )
+
+    def change_primary(self, email_address, user=None):
+        user = user or self.user
+        email_pk = EmailAddress.objects.get(email=email_address).id
+        return self.client.post(
+            reverse('user-change_primary_email', kwargs={'pk': user.id, 'version': 'v1'}),
+            data={'email_pk': email_pk}
+        )
+
+    def test_add_email_permissions(self):
+        email1, email2, email3 = ('email1@example.com', 'email2@example.com', 'email3@example.com')
+        emails = self.user.email_addresses.all().values_list('email', flat=True)
+        assert email1 not in emails and email2 not in emails and email3 not in emails
+        self.login(username='test')
+        resp = self.add_email(email1)
+        assert resp.status_code == 201
+        self.login(username='admin')
+        resp = self.add_email(email2)
+        assert resp.status_code == 201
+        self.login(username='norm')
+        resp = self.add_email(email3)
+        assert resp.status_code == 403
```

Comment:
user `test` can add emails, because he is adding them to himself.
user `admin` can, because he is a super user
user `norm` can't because he is not a super user and is trying to add an email address for someone other than himself.

Reply:
Refactoring this to make it clear that `self.add_email`, `remove_email` and `change_primary` all try to add to `self.user`, not the currently logged in user.

Reply:
could we add comments for this? 

> _Non-superuser cannot change password on another user_

(also re `PermissionDenied`, could/should be asserting some clear message is returned here)